cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -986,7 +987,8 @@
 ;; Vector shifts.
 (rule (lower (has_type (ty_vec128 ty) (ushr x y)))
       (let ((size VectorSize (vector_size ty))
-            (shift Reg (vec_dup (sub $I32 (zero_reg) y) size)))
+            (masked_shift_amt Reg (and_imm $I32 y (shift_mask ty)))
+            (shift Reg (vec_dup (sub $I32 (zero_reg) masked_shift_amt) size)))
```

Comment:
Just a quick comment - I am surprised that passing `$I32` to `sub` works for `I64X2` vectors, since the shift amount should end up being a large positive number (or 0) instead of a small negative number.

Reply:
I'm guessing `dup` will sign extend? That's the only way I can see this working, but I can't find any mention of sign extending in the docs.

I've switched to a 64 bit sub.

Reply:
Ah, now I remember why the original code works and is in fact correct, but I still like the usage of `$I64` - I forgot that the vector shifts used only **the least significant byte** of the shift amount, so there is in fact masking, just not the kind we need.